app/models/miq_server/worker_management/process.rb

```diff
@@ -1,15 +1,28 @@
 class MiqServer::WorkerManagement::Process < MiqServer::WorkerManagement
   def sync_from_system
     require "sys/proctable"
-    self.miq_processes = Sys::ProcTable.ps.select { |proc| proc.ppid == my_server.pid }
+    self.miq_processes        = Sys::ProcTable.ps.select { |proc| proc.ppid == my_server.pid }
+    self.miq_processes_by_pid = miq_processes.index_by(&:pid)
   end
 
   def sync_starting_workers
-    MiqWorker.find_all_starting.to_a
+    sync_from_system
+    starting = MiqWorker.find_all_starting
+    starting.where(:pid => miq_processes_by_pid.keys)
+            .reject(&:rails_worker?)
```

Comment:
Do we have `non_rails_worker?`. It might make more sense to `select` those instead of rejecting the rails ones.  

Is it overkill to have these methods do something like:

```ruby
def sync_starting_workers
  sync_from_system       # get the latest from process list
  sync_rails_workers     # no-op, workers do this on their own
  sync_non_rails_workers # move the new code here - it's an extra query but it's just as easy to update the non-rails worker rows by pid to started
  
  MiqWorker.find_all_starting.to_a
end

Reply:
Is it possible to do rails_worker as a query scope on MiqWorker?  I'm thinking something similar concrete_subclasses where we can get a list of the possible types, then issue a `.where(:type => those_subclasses)`.  That way we can optimize the update to an update_all.

(Not necessary for this PR, but I can see it being more useful when chaining things later)

Reply:
Yeah I started doing this as a scope but I think that'd be better off as a separate PR, also the switch to select(&:non_rails_worker?) because we'd want to change the other classes (systemd/kubernetes) as well.

Reply:
Yeah, I like the scope by worker classes (rails vs non-rails) idea but agree we should do that across the board for other classes.

I'm fine with the existing code if you want to refactor it after to do the inverse `non_rails_worker?` + extract a method.  Whatever is easier to verify.

Reply:
I do like splitting this up into methods so let me get that in

Reply:
Done, PTAL